### ISSUE TITLE
MEFISTO2 trte.f/.c

### ISSUE THREAD

**Issue by andrewfu77** [trelau/SMESH#45]
In extra/MEFISTO2/trte.c, these lines: [1925](https://github.com/trelau/SMESH/blob/master/extra/MEFISTO2/trte.c#L1925) and [1968](https://github.com/trelau/SMESH/blob/master/extra/MEFISTO2/trte.c#L1968) don't seem to match the original source, trte.f line [1297](https://git.salome-platform.org/gitweb/?p=modules/smesh.git;a=blob;f=src/MEFISTO2/trte.f;h=a330bebb9b7bdfb9099d532e7f3924880b64fba3;hb=cca498b7916f338f323c76052b3bb05666cf071a#l1297). I believe this is a mistake, because, as far as I can tell, it looks like the upstream developers mean to invoke [this](https://git.salome-platform.org/gitweb/?p=modules/smesh.git;a=blob;f=src/MEFISTO2/aptrte.cxx;h=807575a51954e6a0e7ff199f904e8c64d8eabd50;hb=cca498b7916f338f323c76052b3bb05666cf071a#l32) version of the function defined in aptrte.cxx which returns a value as opposed to modifying an out-parameter. They seem to also be relying on undefined behavior, passing extra parameters that are not in the declaration.

**Comment by looooo** [comment]
In freecad meshing with mefisto fails with:

```
th. 140131097933056 - Trace ../src/SMESH/src/SMESH/SMESH_subMesh.cxx [2005] : MEFISTO_2D failed on sub-shape #33 with error 2 "Error in Triangulation (aptrte())"
```
not sure if this is related

**Comment by trelau** [comment]
The trte.c file I grabbed from FreeCAD repo. At first I tried to implement my own Fortran to C conversion during the build process, but after some attempts it didn't seem work the effort, so just grabbed the file from FreeCAD.

**Comment by trelau** [comment]
Resolved by #46
